src/app/register-email-form/register-email-form.component.html

```diff
@@ -14,13 +14,16 @@ <h1>{{MESSAGE_PREFIX + '.header'|translate}}</h1>
                     <label class="font-weight-bold"
                            for="email">{{MESSAGE_PREFIX + '.email' | translate}}</label>
                     <input [className]="(email.invalid) && (email.dirty || email.touched) ? 'form-control is-invalid' :'form-control'"
-                           type="text" id="email" formControlName="email"/>
+                           type="text" id="email" formControlName="email"
+                           [attr.aria-label]="MESSAGE_PREFIX + '.aria.label' | translate"
+                           aria-describedby="email-errors-required email-error-not-valid"
```

Comment:
I'm not sure this `aria-describedby` should be hardcoded.  If there are no errors on the form, then both "email-errors-required" and "email-error-not-valid" will **not exist**.  In that scenario, won't this `aria-describedby` be **incorrect**?

It seems like the `aria-describedby` field should be more dynamic.  It should say `email-errors-required` if `email.errors` is true, but it should say `email-error-not-valid` if `email.invalid` is true.

Reply:
Hello, good afternoon @tdonohue!

Certainly if there is no error in the form the “aria-describedby” is not necessary, but in this case, its purpose is to announce to users who make use of assistive technology that there are errors in the form and therefore it is not possible to continue. Like sometimes, we type our email in the wrong way and don't realize it hehe.

Do you think this isn't necessary, or do you know of a better way to address this issue?

I welcome your suggestions. Looking forward to hearing from you!

Reply:
Hi @Andrea-Guevara : The problem that I see is that when there are **no errors**, then ` aria-describedby="email-errors-required email-error-not-valid"` will point at fields that **do not exist in the page**.   In other words, the `email-errors-required` field ONLY exists if the email is empty.    And, the `email-error-not-valid` field ONLY exists if an invalid email is entered.

So, it seems like the `aria-describedby` should ONLY point at fields which exist.  If an error field is visible, then it should be listed in the `aria-describedby`.  But, if there are no errors, then the `aria-describedby` should be empty (or not exist).

Reply:
If it's not clear, I agree that we need the `aria-describedby` attribute to notify users of errors in the form.  But, I feel that attribute should *not exist* when there are no errors in the form (otherwise, it is notifying users of something that doesn't exist).  The `aria-describedby` also should only ever have one value (at most), because *either* the `email-errors-required` error will exist, *or* the `email-error-no-valid` error will exist (but not both at the same time).

Reply:
Hello @tdonohue

Thank you very much for your feedback. I understand your comment! I've provided a possible fix

I look forward to hearing from you